docker/test/upgrade/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# rebuild in #33610
+# docker build -t clickhouse/upgrade-check .
+ARG FROM_TAG=latest
+FROM clickhouse/stateful-test:$FROM_TAG
```

Comment:
This must be reflected in `docker/images.json` for `clickhouse/stateful-test`, otherwise it is error prone

Reply:
Thank you for noticing! Let me fix it

Reply:
https://github.com/ClickHouse/ClickHouse/pull/47031